### PR TITLE
Fix partial reconnect issues

### PR DESCRIPTION
**Pull request by oleks-rip**

### CODE REVIEW

File: src/xbwd/client/ChainListener.cpp [hunk 1]
```diff
@@ -1442,11 +1488,7 @@ ChainListener::processAccountTxHlp(Json::Value const& msg)
     }
 
     auto const& transactions = result[ripple::jss::transactions];
-    if (!transactions.size())
```
Comment (pwang200) [inline src/xbwd/client/ChainListener.cpp:1445]: why is this check removed? is it replaced by this check above?
```
     if (ripple::RPC::contains_error(msg))
￼    {
￼        // request this ledger once more
￼        warnMsg("error in msg");
￼        return false;
￼    }

File: src/xbwd/client/ChainListener.cpp [hunk 2]
```diff
@@ -63,6 +64,11 @@ ChainListener::ChainListener(
     , signingAccount_(signingAccount)
     , j_{j}
 {
+    inRequest_ = false;
+    ledgerReqMax_ = 0;
+    prevLedgerIndex_ = 0;
+    txnHistoryIndex_ = 0;
+    hp_.clear();
```
Comment (seelabs) [inline src/xbwd/client/ChainListener.cpp:71]: Shouldn't `hp_` start in a cleared state? Why do we need to clear here explicitly?

File: src/xbwd/client/ChainListener.cpp [hunk 3]
```diff
@@ -63,6 +64,11 @@ ChainListener::ChainListener(
     , signingAccount_(signingAccount)
     , j_{j}
 {
+    inRequest_ = false;
+    ledgerReqMax_ = 0;
+    prevLedgerIndex_ = 0;
+    txnHistoryIndex_ = 0;
```
Comment (seelabs) [inline src/xbwd/client/ChainListener.cpp:70]: These should already have the correct values, why are we setting them explicitly?

File: src/xbwd/client/ChainListener.h [hunk 4]
```diff
@@ -99,6 +99,8 @@ class ChainListener : public std::enable_shared_from_this<ChainListener>
     bool inRequest_ = false;
     // last ledger requested for new tx
     unsigned ledgerReqMax_ = 0;
+    // last ledger that was processed (in case of errors / disconnects)
+    unsigned ledgerProcessedDoor_ = 0, ledgerProcessedSign_ = 0;
```
Comment (seelabs) [inline src/xbwd/client/ChainListener.h:103]: Rippled puts member vars on their own line, even when they are the same type. We should probably follow that style as well.

**Review by seelabs — COMMENTED** [review]
Left some minor comments.

File: src/xbwd/client/ChainListener.cpp (hunk 1 above)
Comment (oleks-rip) [reply src/xbwd/client/ChainListener.cpp:1445]: It was optional here, now I've change return value and just remove on point of exit, to not to duplicate code.
I'll add more comments

Comment (oleks-rip) [reply src/xbwd/client/ChainListener.cpp:1445]: fixed 64de4766d2

File: src/xbwd/client/ChainListener.cpp (hunk 2 above)
Comment (oleks-rip) [reply src/xbwd/client/ChainListener.cpp]: fixed 64de4766d2

File: src/xbwd/client/ChainListener.cpp (hunk 3 above)
Comment (oleks-rip) [reply src/xbwd/client/ChainListener.cpp]: fixed 64de4766d2

File: src/xbwd/client/ChainListener.h (hunk 4 above)
Comment (oleks-rip) [reply src/xbwd/client/ChainListener.h]: fixed 64de4766d2

File: src/xbwd/client/ChainListener.cpp (hunk 1 above)
Comment (pwang200) [reply src/xbwd/client/ChainListener.cpp:1445]: Olek explained to me in DM.

**Review by seelabs — APPROVED** [review]

**Comment by seelabs** [comment]
merged

**Pull request closed without merging**